src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala

```diff
@@ -52,6 +61,42 @@ class DetectedLanguageSuitev4 extends TestBase with DataFrameEquality with TextK
       iso(1).get(0).toString == "fr" && iso(1).get(1).toString == "zh")
   }
 
+  test("Language Detection - Invalid Document Input"){
+    val replies = detector.transform(invalidDocDf)
+      .select("output.error.errorMessage")
+      .collect()
+    val errors = replies.map(row => row.getList(0))
+    assert(errors(0).get(0).toString == "Document text is empty."
+      && errors(0).get(1).toString == "Document text is empty.")
+  }
+
+  test("Invalid Subscription Key Caught") {
+    val invalidKey = "12345"
+    val caught =
+      intercept[SparkException] {
+        invalidDetector
+          .setEndpoint("https://eastus.api.cognitive.microsoft.com/")
+          .setSubscriptionKey(invalidKey)
+          .transform(df)
+          .show()
+      }
+    assert(caught.getMessage.contains("Status code 401") &&
+      caught.getMessage.contains("invalid subscription key or wrong API endpoint"))
+  }
+
+  test("Wrong API Endpoint Caught") {
+    val invalidEndpoint = "invalidendpoint"
+    val caught =
+      intercept[SparkException] {
+        invalidDetector
+          .setEndpoint(invalidEndpoint)
+          .setSubscriptionKey(textKey)
+          .transform(df)
+          .show()
+      }
+    assert(caught.getMessage.contains("'endpoint' must be a valid URL"))
+  }

```

Comment:
For other endpoints such as sentiment/kpe, test case could also cover the invalid language support.